### PR TITLE
feature/ads-794 Present carbon data

### PR DESCRIPTION
**Pull request by APJohns**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -151,6 +153,11 @@ const App = (): JSX.Element => {
 
               <Route path="search" element={<SearchResults />} />
 
+              <Route path="carbon-metrics">
+                <Route path="" element={<CarbonMetrics />} />
+                <Route path="filter" element={<PageMetrics />} />
+              </Route>
```
Comment (kaseyllabelle) [inline src/App.tsx:159]: we may need an aria-live region if we update page content w/o rerouting

File: src/pages/carbonMetrics/CarbonMetrics.tsx [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+import SparkChart from './SparkChart';
+
+export interface Averages {
+  date: string;
+  carbon: number;
+  percent: number;
+}
+
+const CarbonMetrics = (): JSX.Element => {
+  useTitle({ titlePrefix: `Carbon Metrics` });
```
Comment (kaseyllabelle) [inline src/pages/carbonMetrics/CarbonMetrics.tsx:17]: sentence case :)

File: src/pages/carbonMetrics/CarbonMetrics.tsx [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+        carbon: 0,
+        percent: 0
+      };
+    }
+  };
+
+  return (
+    <Layout>
+      <Helmet>
+        <meta name="description" content="Boston Scientific Anatomy Design System website carbon metrics" />
```
Comment (kaseyllabelle) [inline src/pages/carbonMetrics/CarbonMetrics.tsx:127]: "Anatomy Design System's carbon metrics"?

File: src/pages/carbonMetrics/CarbonMetrics.tsx [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+      </Helmet>
+      <div className="docs-body-minimal">
+        <main id="mainContent">
+          <div className="docs-page-header">
+            <div className="docs-metadata">
+              <h1 className="docs-title">Carbon metrics</h1>
+              {!!latestData && (
+                <dl className="docs-datestamp">
+                  <dt>Last Updated:</dt>
```
Comment (kaseyllabelle) [inline src/pages/carbonMetrics/CarbonMetrics.tsx:136]: i just noticed we're doing title case for this everywhere. we can update later.

File: src/pages/carbonMetrics/CarbonMetrics.tsx [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+                    <p className="docs-metrics-tile-change">
+                      {getLatestChanges().percent >= 0 && '+'}
+                      {getLatestChanges().percent}%
+                    </p>
+                    <SparkChart data={averages} yValue="percent" />
+                  </div>
+                </div>
+              )}
+              <h3>Pages</h3>
+              <table className="docs-table-responsive">
```
Comment (kaseyllabelle) [inline src/pages/carbonMetrics/CarbonMetrics.tsx:173]: can't believe you voluntarily built a sortable table 😅

File: src/pages/carbonMetrics/CarbonMetrics.tsx [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+                <caption className="bsds-visually-hidden">
+                  Most recent carbon data for each page across the Anatomy documentation site.
+                </caption>
+                <thead>
+                  <tr>
+                    <th className="docs-table-header-sortable" aria-sort={getSortAria('url')}>
```
Comment (kaseyllabelle) [inline src/pages/carbonMetrics/CarbonMetrics.tsx:179]: can we check w3c patterns for sorting
https://www.w3.org/WAI/ARIA/apg/patterns/grid/

we can make changes in a future PR or ticket if needed

File: src/pages/carbonMetrics/PageMetrics.tsx [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+    if (location.search !== currentParams) {
+      getData();
+      setCurrentParams(location.search);
+    }
+  }, [currentParams, location.search, searchParams]);
+
+  return (
+    <Layout>
+      <Helmet>
+        <meta name="description" content="Boston Scientific Anatomy Design System website carbon metrics" />
```
Comment (kaseyllabelle) [inline src/pages/carbonMetrics/PageMetrics.tsx:92]: same here

File: src/styles/pages/_carbon-metrics.scss [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+  }
+}
+
+.docs-carbon-filters {
+  display: flex;
+
+  dd {
+    margin-left: 1ch;
+  }
+}
```
Comment (kaseyllabelle) [inline src/styles/pages/_carbon-metrics.scss:20]: is this something we can make more abstract and share?

File: src/styles/pages/_carbon-metrics.scss [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+  padding: space(3x);
+  text-align: center;
+
+  .docs-metrics-trend-up {
+    fill: $feedback-success-40;
+  }
+
+  .docs-metrics-trend-down {
+    fill: $feedback-error-40;
+  }
```
Comment (kaseyllabelle) [inline src/styles/pages/_carbon-metrics.scss:42]: do we need these anymore?

File: src/styles/pages/_carbon-metrics.scss [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+
+  h3 {
```
Comment (kaseyllabelle) [inline src/styles/pages/_carbon-metrics.scss:44]: can we use the mt-0 util instead?

File: src/styles/pages/_carbon-metrics.scss [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+    margin-top: 0;
+  }
+}
+
+.docs-metrics-tile-description {
+  @include font-size($fs-body-subtle);
+  margin-bottom: auto;
```
Comment (kaseyllabelle) [inline src/styles/pages/_carbon-metrics.scss:51]: is there a bottom margin coming from something?

File: src/styles/pages/_carbon-metrics.scss [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+  justify-content: center;
+  align-items: center;
+  gap: space(sm);
+}
+
+.docs-spark-chart {
+  display: block;
+  height: auto;
+  max-width: 100%;
+  margin: space(3x) auto 0 auto;
```
Comment (kaseyllabelle) [inline src/styles/pages/_carbon-metrics.scss:69]: can cut the last auto

File: src/styles/pages/_carbon-metrics.scss [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+
+  .line {
+    stroke: $action-primary-40;
+  }
+
+  text {
+    @include font-size($fs-body-subtle);
+  }
```
Comment (kaseyllabelle) [inline src/styles/pages/_carbon-metrics.scss:77]: are we using this in the chart?

File: src/pages/carbonMetrics/CarbonMetrics.tsx (hunk 5 above)
Comment (APJohns) [reply src/pages/carbonMetrics/CarbonMetrics.tsx:173]: I also was shocked I made that decision. It was the best solution for the job 😅

File: src/styles/pages/_carbon-metrics.scss (hunk 11 above)
Comment (APJohns) [reply src/styles/pages/_carbon-metrics.scss:51]: This pushes the flex items below it down so that things line up nice when text wraps.

Comment (kaseyllabelle) [reply src/styles/pages/_carbon-metrics.scss:31]: thoughts on adding a code comment explaining this?

**Review by kaseyllabelle — APPROVED** [review]

**Pull request merged**